media/js/src/chat.js

```diff
@@ -94,40 +68,46 @@ $(function() {
         entry.append('<div class="col-md-2 nick"><a href="' + data.userURL +
                      '">' + data.fullname + '</a></div>');
         var attr = 'message_text';
-        entry.append('<div class="col-md-9 ircmessage">' + data[attr] +
-                     '</div>');
+        entry.append('<div class="col-md-9 ircmessage">' +
+                     renderer.render(data[attr]) + '</div>');
         appendLog(entry);
     };
 
-    function appendLog(msg) {
+    var appendLog = function(msg) {
         var d = log[0];
         var doScroll = d.scrollTop === d.scrollHeight - d.clientHeight;
         msg.appendTo(log);
         if (doScroll) {
             d.scrollTop = d.scrollHeight - d.clientHeight;
         }
-    }
+    };
 
-    $('#msg_form').submit(function() {
-        var msg = $('#text-input');
-        $.ajax({
-            type: 'POST',
-            url: window.postURL,
-            data: {'text': msg.val()},
-            success: function() {
-                msg.val('');
-            },
-            error: function() {
-                alert('post failed');
-            }
+    domReady(function() {
+        if ($('#msg_form').length < 1) {
+            return;
+        }
```

Comment:
I guess it bothers me that with this approach, the chat code is now loaded on every single page of the site (since `main.js` comes in on `base.html` and then just does nothing if a particular element isn't present. It seems wasteful to me since there's only the one view that uses it.